### ISSUE TITLE
Should we exclude `rel=nofollow` by default?

### ISSUE THREAD

**Issue by tunetheweb** [WICG/nav-speculation#309]
We've discovered that WooCommerce "add to basket" buttons are regular links. Now of course that breaks all sorts of GETs should be idempotent but such is the web.

They do implement a `rel=nofollow` to prevent search engines following such links. We're going to add it to our WordPress plugin, but I wonder if that's something we should do by default since it seems like a good sign that it's unexpected to be followed by bots and only by real humans.

Not sure if this is a spec question, or should be up to individual implementers so raising here first.

**Comment by tunetheweb** [comment]
Relevant WordPress issue: https://github.com/WordPress/performance/issues/1140

**Comment by domenic** [comment]
I like this idea in theory. But I can't think of how we would reasonably specify it or explain it. "You said to prerender 'all links matching the `/*` URL pattern', but what we _believe_ you meant was 'all links matching the `/*` URL pattern and without `rel="nofollow"`'. We'd need some concept of... default conditions which get `and`ed onto any that are supplied?? And some way of undoing those???

Maybe it's best to just take this as good advice for when we create examples and documentation? Not sure.

**Comment by domenic** [comment]
> In the same way that `"href_matches": "*"` doesn't match external links for document rules even thought you said `*` (i.e. everything).

It does match :)

**Comment by tunetheweb** [comment]
> It does match :)

I'm confused.

This example: https://prerender-demos.glitch.me/document-rules.html has this:

```html
  <p><a href="next.html">This link should be eligible to be prerendered</a></p>
  <p><a href="next2.html">This link should also be eligible to be prerendered</a></p>
  <p><a href="https://www.example.com">This link is external and should not be eligible</a></p>

  <script type="speculationrules">
    {
      "prerender": [
        {
          "where": {"href_matches":"*"},
          "eagerness": "moderate"
        }
      ]
    }
  </script>
```

But shows this in devtools:

<img width="637" alt="image" src="https://github.com/WICG/nav-speculation/assets/10931297/d2f54c19-ba03-4577-bfe0-1762e1d00342">

So the external link is excluded from the list of potential speculations? And that makes sense to me as the URL pattern has a base of the document AFAIK.

**Comment by domenic** [comment]
You're right, that's because `*` doesn't mean everything, it means URLs which match the current URL plus a `*` segment at the end.

**Comment by domenic** [comment]
Issue triage: let me close this, as it feels too inelegant to include.
